src/ProcessMaker/Nayra/Bpmn/Models/StandardLoopCharacteristics.php

```diff
@@ -0,0 +1,265 @@
+        TransitionInterface $source
+    ) {
+        $item = $this->getInputDataItemValue($instance, $loopCounter);
+        $properties['data'] = [];
+        $properties['data'] = array_merge($properties['data'], (array) $item);
+        $properties['data']['loopCounter'] = $loopCounter;
+        $newToken = $nextState->createToken($instance, $properties, $source);
+        $this->setLoopInstanceProperty($newToken, 'loopCounter', $loopCounter);
+        $this->setLoopInstanceProperty($newToken, 'nextState', $nextState);
+        $this->setLoopInstanceProperty($newToken, 'source', $source);
+        $nextState->addToken($instance, $newToken, false, $source);
+    }
+
+    /**
+     * @param ExecutionInstanceInterface $instance
+     * @param TokenInterface $token
+     *
+     * @return bool
+     */
+    public function continueLoop(ExecutionInstanceInterface $instance, TokenInterface $token)
+    {
+        $continue = $this->checkAfterLoop($instance, $token);
+        if ($continue) {
+            $properties = $this->prepareLoopInstanceProperties($token, []);
+            // LoopInstance Counter
+            $loopCounter = $this->getLoopInstanceProperty($token, 'loopCounter', 0);
+            // Token loopCounter
+            $tokenLoopCounter = $token->getProperty('data', [])['loopCounter'] ?? 0;
+            if ($loopCounter === $tokenLoopCounter) {
+                $loopCounter++;
+                $this->createInstance(
+                    $instance,
+                    $properties,
+                    $loopCounter,
+                    $this->getLoopInstanceProperty($token, 'nextState'),
+                    $this->getLoopInstanceProperty($token, 'source')
+                );
+            }
+        }
+        return $continue;
+    }
+
+    /**
+     * @param ExecutionInstanceInterface $instance
+     * @param TokenInterface $token
+     * @return bool
+     */
+    public function isLoopCompleted(ExecutionInstanceInterface $instance, TokenInterface $token)
+    {
+        $testBefore = $this->getTestBefore() ?: true;
+        $condition = $this->getLoopCondition();
+        $data = $instance->getDataStore()->getData();
+        $evaluatedCondition = $condition($data);
+        $loopMaximum = $this->getLoopMaximumFormalExpression($data);
+        $loopCounter = $this->getLoopInstanceProperty($token, 'loopCounter', 0);
+        $loopCondition = $loopMaximum === null || $loopCounter < $loopMaximum;
+        if ($testBefore && $loopCounter === 0) {
+            return false;
+        }
+        if ($testBefore && $evaluatedCondition && $loopCondition) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check before the loop should be executed
+     *
+     * @param  ExecutionInstanceInterface $instance
+     * @param  TokenInterface $token
+     * @return void
+     */
+    public function checkBeforeLoop(ExecutionInstanceInterface $instance, TokenInterface $token)
```

Comment:
Unused method, please add it to test before condition

Reply:
Looks like this method could be private, please change it to private or if it is needed by another class add it to the interface declaration